src/core/batch.rkt

```diff
@@ -96,10 +57,77 @@
       (unmunge root)))
   exprs)
 
+(define (expand-taylor batch)
+  (define vars (batch-vars batch))
+  (define icache (reverse vars))
+  (define exprhash
+    (make-hash (for/list ([var vars] [i (in-naturals)])
+                 (cons var i))))
+  ; Counts
+  (define size 0)
+  (define exprc 0)
+  (define varc (length vars))
+
+  ; Translates programs into an instruction sequence of operations
+  (define (munge prog)
```

Comment:
After a long attempt of write this function as an update to `nodes`, I decided to simply convert `nodes` to an `expr` once again and `munge` for the second time.. Super expensive, a cheaper algorithm should be created